### PR TITLE
WIP: Runner Incoming Data Receiver

### PR DESCRIPTION
**Pull request by matburt**
Ansible Runner has the ability to execute and orchestrate jobs. This adds an interface that lets AWX consume and display those without specifically requiring the linkage to other AWX resources

Some things left to be done:

- [ ] connection security
- [ ] better failure handling
- [ ] testing
- [ ] packaging work
- [ ] notifications(?) and logging(?)

### CODE REVIEW

File: awx/main/management/commands/runner_receiver.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class Command(BaseCommand):
+
+    help = 'Launch a receiver for Runner Events and Statuses'
+
+    def handle(self, *args, **kwargs):
+        loop = asyncio.get_event_loop()
+        proto = RunnerClientProtocol(loop)
+        listener = loop.create_server(lambda: proto,
```
Comment (ryanpetrello) [inline awx/main/management/commands/runner_receiver.py:57]: Using https://docs.python.org/3/library/asyncio-stream.html#asyncio.start_unix_server would probably go a long way for the `security` checkbox (by using a domain socket we could restrict via file permissions).

**Comment by shanemcd** [comment]
@matburt  As I'm getting more familiar with this part of the codebase I may have some stupid questions. This may be one of them. I am reading through and learning about how @ryanpetrello's dispatcher code works. Would it be prudent to reuse that pattern here? `AWXConsumer` takes care of managing a worker pool which may be important here too? Another thing I noticed is it looks like `CallbackBrokerWorker` does some extra things before persisting events to the database. Do we want to figure out how to get these events into that codepath? Let me know if this does or doesn't make sense.

File: awx/main/management/commands/runner_receiver.py (hunk 1 above)
Comment (matburt) [reply awx/main/management/commands/runner_receiver.py:57]: Well the problem with that is that it implies the Runner service will always be local which will almost universally not be true.

**Comment by matburt** [comment]
asyncio implements a pretty dramatically different pattern than other types of network services. It's the same thing you'll see in systems the use libraries like libevent. In this case there isn't a strict *need* to do something like multiprocess connection pooling in the traditional sense. asyncio as a pattern can track a lot of connections and activity in a single thread pretty efficiently (thousands of requests per second) without the overhead of forking the python process or the complexity of manage multi-process IPC. We also have the ability to swap out the lower level event loop implementation if we *really* need to improve efficiency.

It *is* entirely possible that some of this logic needs to be brought into the event processor here (like the notification hooks). I *am* hoping we can dramatically simplify incoming event processing under this event service since this will ultimately become *the* interface for external executors (Runner) and we won't have a need to use the rabbitmq-bound callback receiver system anymore.

File: tools/docker-compose/Dockerfile [hunk 2]
```diff
@@ -17,7 +17,7 @@ RUN yum -y update && yum -y install openssh-server ansible mg vim tmux \
   xmlsec1-devel swig krb5-devel xmlsec1-openssl xmlsec1 \
   xmlsec1-openssl-devel libtool-ltdl-devel rabbitmq-server bubblewrap \
   zanata-python-client gettext gcc-c++ libcurl-devel bzip2 \
-  rsync
+  rsync zeromq
```
Comment (shanemcd) [inline tools/docker-compose/Dockerfile]: Did you end up actually using zeromq for this?

**Comment by chrismeyersfsu** [comment]
What's the auth mechanism for this?

**Comment by matburt** [comment]
SSL Certificates

File: tools/docker-compose/Dockerfile (hunk 2 above)
Comment (matburt) [reply tools/docker-compose/Dockerfile]: nah, replaced this with asyncio so this is unneeded

**Comment by ryanpetrello** [comment]
@matburt I'm cleaning up PRs.  Given its age, do you want to keep this open in its current state, or close and revisit later?

**Comment by matburt** [comment]
Lets keep this one open.

**Pull request closed without merging**